core/charm/downloader/downloader.go

```diff
@@ -227,26 +226,18 @@ func (d *Downloader) storeCharm(charmURL *charm.URL, dc DownloadedCharm, archive
 }
 
 func (d *Downloader) normalizePlatform(charmURL *charm.URL, platform corecharm.Platform) (corecharm.Platform, error) {
-	os := platform.OS
-	if platform.Series != "" {
-		sys, err := series.GetOSFromSeries(platform.Series)
```

Comment:
not sure about removing all these checks to ensure that the OS is set.

Reply:
Tracing through the code, this method is only called with output from `DeduceOrigin`, where everything is correctly set up.
We are moving off series - 3.0 will not have it - so need to remove dependencies on it. There's no need to do the check here as it's already been done upstream.We need to consolidate any series dependencies to the smallest footprint so they can be dealt with with less churn in 3.0